soc/arm/gigadevice/Kconfig

```diff
@@ -15,6 +15,11 @@ config SOC_FAMILY_GD32_ARM
 	bool
 	select SOC_FAMILY_GD32
 
+config GD32_HAS_AF_PINMUX
+	bool
+	help
+	  This option should be selected if the series use an AF pinmux model.
+
```

Comment:
I think it is better move to modules/hal_gigadevice. That way we can share between ARM and RISC-V

Reply:
done